pkg/resource/imageregistryca.go

```diff
@@ -0,0 +1,187 @@
+		openshiftConfigLister:     openshiftConfigLister,
+		serviceLister:             serviceLister,
+		imageRegistryConfigLister: imageRegistryConfigLister,
+		storageListers:            storageListers,
+		kubeconfig:                kubeconfig,
+		client:                    client,
+	}
+}
+
+func (gcac *generatorImageRegistryCA) Type() runtime.Object {
+	return &corev1.ConfigMap{}
+}
+
+func (gcac *generatorImageRegistryCA) GetNamespace() string {
+	return defaults.OpenShiftConfigManagedNamespace
+}
+
+func (gcac *generatorImageRegistryCA) GetName() string {
+	return defaults.ImageRegistryCAName
+}
+
+func (gcac *generatorImageRegistryCA) storageDriver() (storage.Driver, bool, error) {
+	imageRegistryConfig, err := gcac.imageRegistryConfigLister.Get("cluster")
+	if errors.IsNotFound(err) {
+		return nil, false, nil
+	} else if err != nil {
+		return nil, false, err
+	}
+
+	if imageRegistryConfig.Spec.ManagementState == operatorv1.Removed {
+		// The certificates controller does not need to know about
+		// storage when the management state is Removed.
+		return nil, false, nil
+	}
+
+	driver, err := storage.NewDriver(&imageRegistryConfig.Spec.Storage, gcac.kubeconfig, gcac.storageListers)
+	if err == storage.ErrStorageNotConfigured || storage.IsMultiStoragesError(err) {
+		return nil, false, nil
+	} else if err != nil {
+		return nil, false, err
+	}
+
+	canRedirect := !imageRegistryConfig.Spec.DisableRedirect
+
+	return driver, canRedirect, nil
+}
+
+func (gcac *generatorImageRegistryCA) expected() (runtime.Object, error) {
+	cm := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      gcac.GetName(),
+			Namespace: gcac.GetNamespace(),
+		},
+		Data:       map[string]string{},
+		BinaryData: map[string][]byte{},
+	}
+
+	var ownHostnameKeys []string
+
+	serviceCA, err := gcac.lister.Get(defaults.ServiceCAName)
+	if errors.IsNotFound(err) {
+		klog.V(4).Infof("missing the service CA configmap: %s", err)
+	} else if err != nil {
+		return cm, err
+	} else {
+		if cert, ok := serviceCA.Data["service-ca.crt"]; ok {
+			internalHostnames, err := getServiceHostnames(gcac.serviceLister, defaults.ServiceName)
+			if err != nil {
+				return cm, err
+			}
+			if len(internalHostnames) == 0 {
+				klog.Infof("unable to get the service name to add service-ca.crt")
+			} else {
+				for _, internalHostname := range internalHostnames {
+					key := strings.Replace(internalHostname, ":", "..", -1)
+					ownHostnameKeys = append(ownHostnameKeys, key)
+					cm.Data[key] = cert
+				}
+			}
+		} else {
+			klog.Infof("the service CA is not injected yet")
+		}
+	}
+
+	driver, canRedirect, err := gcac.storageDriver()
+	if err != nil {
+		return cm, err
+	}
+	if driver != nil {
+		storageCABundle, _, err := driver.CABundle()
+		if err != nil {
+			return cm, err
+		}
+		if storageCABundle != "" {
+			klog.V(4).Infof("using storage ca bundle (%d bytes)", len(storageCABundle))
+			cm.Data["storage-ca-bundle.pem"] = storageCABundle
```

Comment:
afaik we need it only for image-registry so that it always trusts the storage certs; clients don't need `storage-ca-bundle.pem`

Reply:
makes sense - will remove.